### PR TITLE
Feature/issued credentials api

### PR DESCRIPTION
**Pull request by jstuczyn**
# Description

Closes: NC-213
Closes: BN-11

### CODE REVIEW

File: common/nymcoconut/src/elgamal.rs [hunk 1]
```diff
@@ -112,7 +115,16 @@ impl Bytable for PrivateKey {
     }
 
     fn try_from_byte_slice(slice: &[u8]) -> Result<Self> {
-        PrivateKey::from_bytes(slice.try_into().unwrap())
+        let received = slice.len();
+        let Ok(arr) = slice.try_into() else {
+            return Err(CoconutError::UnexpectedArrayLength {
+                typ: "elgamal::PrivateKey".to_string(),
```
Comment (octol) [inline common/nymcoconut/src/elgamal.rs:121]: is "typ" a typo for "type", or is it an abbreviation for something?

File: nym-api/nym-api-requests/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
-// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: GPL-3.0-only
```
Comment (octol) [inline nym-api/nym-api-requests/src/lib.rs]: Do we want this for something that other crates are likely to import?

File: nym-api/nym-api-requests/src/models.rs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
-// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: GPL-3.0-only
```
Comment (octol) [inline nym-api/nym-api-requests/src/models.rs]: Same as other comment

File: nym-api/src/coconut/error.rs [hunk 4]
```diff
@@ -30,22 +30,32 @@ pub enum CoconutError {
     #[error(transparent)]
     SerdeJsonError(#[from] serde_json::Error),
 
-    #[error("Could not parse Ed25519 data - {0}")]
+    #[error("Could not parse Ed25519 data: {0}")]
```
Comment (octol) [inline nym-api/src/coconut/error.rs]: nit: error strings start in lowercase

File: nym-api/src/support/identity.rs [hunk 5]
```diff
@@ -0,0 +1,2 @@
+// Copyright 2023 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
```
Comment (octol) [inline nym-api/src/support/identity.rs]: hmm other nym-api stuff was  GPL?

**Review by octol — APPROVED** [review]

File: common/nymcoconut/src/elgamal.rs (hunk 1 above)
Comment (jstuczyn) [reply common/nymcoconut/src/elgamal.rs:121]: it's not a typo - we can't use `type` as it's a reserved keyword. so similarly to some other projects, we're using `typ` instead.

**Pull request merged**